### ISSUE TITLE
Routing & Nonce Required

### ISSUE THREAD

**Issue by fritterhoff** [nibtime/next-safe-middleware#77]
I'm trying to use this library in a project where several (sub-)pages require a nonce for inline styles. In case of loading the subpages directly the nonce is returned correctly and loading the pages works fine. In case of loading a statically rendered page before the server rendered page loading and applying the inline styles fails due to the lack of the nonce in the CSP header.

Is there (maybe) an option to enforce includint the nonce in the initial request, even it is not used directly? Due to the next.js routing the subpages are not "loaded" the same way as navigating to them with the direct URL...

To sum up:

1. Loading a static rendered page: header does not include a nonce
2. Loading a server rendered page via router "button" no new header is returned: nonce is missing and inline style fails

![grafik](https://user-images.githubusercontent.com/32478819/194770460-ffb4460b-e3b3-41ac-b216-2f0a80a883f3.png)

**Comment by fritterhoff** [comment]
I've just build a mini showcase using MUI and the MUI datagrid: https://github.com/fritterhoff/csp-mui
